packages/@aws-cdk/aws-stepfunctions-tasks/lib/ecs/run-task.ts

```diff
@@ -0,0 +1,376 @@
+        NetworkConfiguration: this.networkConfiguration,
+        Overrides: renderOverrides(this.props.containerOverrides),
+        ...this.props.launchTarget.bind(this).parameters,
+      }),
+    };
+  }
+
+  private configureAwsVpcNetworking() {
+    const subnetSelection = this.props.subnets ?? { subnetType: this.props.assignPublicIp ? ec2.SubnetType.PUBLIC : ec2.SubnetType.PRIVATE };
+
+    this.networkConfiguration = {
+      AwsvpcConfiguration: {
+        AssignPublicIp: this.props.assignPublicIp ? (this.props.assignPublicIp ? 'ENABLED' : 'DISABLED') : undefined,
+        Subnets: this.props.cluster.vpc.selectSubnets(subnetSelection).subnetIds,
+        SecurityGroups: cdk.Lazy.listValue({ produce: () => [this.securityGroup!.securityGroupId] }),
+      },
+    };
+  }
+
+  private validateLaunchTarget() {
+    if (this.props.launchTarget instanceof EcsFargateLaunchTarget) {
+      this.validateFargateLaunchType();
+    }
+
+    if (this.props.launchTarget instanceof EcsEc2LaunchTarget) {
+      this.validateEc2LaunchType();
+    }
+
+    if (!this.props.taskDefinition.defaultContainer) {
+      throw new Error('A TaskDefinition must have at least one essential container');
+    }
+  }
+
+  private validateEc2LaunchType() {
+    if (!this.props.cluster.hasEc2Capacity) {
+      throw new Error('Cluster for this service needs Ec2 capacity. Call addXxxCapacity() on the cluster.');
+    }
+
+    if (this.props.taskDefinition.networkMode === ecs.NetworkMode.AWS_VPC) {
+      this.configureAwsVpcNetworking();
+    } else {
+      // Either None, Bridge or Host networking. Copy SecurityGroup from ASG.
+      this.validateNoNetworkingProps();
+      this.connections.addSecurityGroup(...this.props.cluster.connections.securityGroups);
+    }
+  }
+
+  private validateNoNetworkingProps() {
+    if (this.props.subnets !== undefined || this.props.securityGroup !== undefined) {
+      throw new Error('vpcPlacement and securityGroup can only be used in AwsVpc networking mode');
+    }
+  }
+
+  private validateFargateLaunchType() {
+    if (!this.props.taskDefinition.isFargateCompatible) {
+      throw new Error('Supplied TaskDefinition is not configured for compatibility with Fargate');
+    }
+
+    this.configureAwsVpcNetworking();
+  }
+
+  private makePolicyStatements(): iam.PolicyStatement[] {
+    const stack = cdk.Stack.of(this);
+
+    // https://docs.aws.amazon.com/step-functions/latest/dg/ecs-iam.html
+    const policyStatements = [
+      new iam.PolicyStatement({
+        actions: ['ecs:RunTask'],
+        resources: [this.props.taskDefinition.taskDefinitionArn],
+      }),
+      new iam.PolicyStatement({
+        actions: ['ecs:StopTask', 'ecs:DescribeTasks'],
+        resources: ['*'],
+      }),
+      new iam.PolicyStatement({
+        actions: ['iam:PassRole'],
+        resources: cdk.Lazy.listValue({ produce: () => this.taskExecutionRoles().map((r) => r.roleArn) }),
```

Comment:
Why lazy? I can't see anything in taskExecutionRoles() that is adjusted  out of band.

Reply:
not really something I've changed, but I'll dig up the rationale behind the original decision. What's the recommendation and why?

Reply:
why use lazy when it's not needed?

```suggestion
        resources: this.taskExecutionRoles().map((r) => r.roleArn),
```

Reply:
you're absolutely right, dropped it